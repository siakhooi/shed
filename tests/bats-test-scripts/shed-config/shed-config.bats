setup(){
    load '../common-setup'
    common_setup
}
@test "shed-config" {
    run shed-config
    assert_success
    cat_expected "shed-config" | assert_output -
}
@test "shed config" {
    run shed config
    assert_success
    cat_expected "shed-config" | assert_output -
}
@test "unset SHED_CONFIG_FILE; shed-config" {
    unset SHED_CONFIG_FILE
    run shed-config
    assert_failure 1
    cat_expected "shed-config-file-not-set" | assert_output -
}
@test "unset SHED_CONFIG_FILE; shed config" {
    unset SHED_CONFIG_FILE
    run shed config
    assert_failure 1
    cat_expected "shed-config-file-not-set" | assert_output -
}
@test "SHED_CONFIG_FILE=/xxx; shed-config" {
    SHED_CONFIG_FILE=/xxx
    run shed-config
    assert_failure 2
    cat_expected "shed-config-file-not-found" | assert_output -
}
@test "SHED_CONFIG_FILE=/xxx; shed config" {
    SHED_CONFIG_FILE=/xxx
    run shed config
    assert_failure 2
    cat_expected "shed-config-file-not-found" | assert_output -
}
