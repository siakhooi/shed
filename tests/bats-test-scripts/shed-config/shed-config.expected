sheds:
  - code: D
    name: shed-D-development
    kubeconfig:
      - context: kube-dev-1
        namespace: ns-application
        prompt-name: dev-app
      - context: kube-dev-1
        namespace: ns-database
  - code: T
    name: shed-T-testing
    kubeconfig:
      - context: kube-test-1
shed-init:
  shed: D
